src/test/java/com/vinyl/AddProductToStoreTest.java

```diff
@@ -0,0 +1,75 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class AddProductToStoreTest extends ManagerBaseIntegration {
+
+    @Autowired
+    protected TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    protected TokenRepository tokenRepository;
+
+    @Test
+    public void testWhenOK() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+    }
+
+    @Test
+    public void testWhenProductNameIsInvalid() {
+        request.setProductName("");
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenProductPriceIsInvalid() {
+        request.setPrice(-1.2);
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenProductStockIsInvalid() {
+        request.setStock(0);
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenTokenIsMissing() {
+        ResponseEntity<?> response = trt.exchange("/products", HttpMethod.POST, new HttpEntity<>(request), Void.class);
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+    }
+
+    @Test
+    public void testWhenTokenIsInvalid() {
+        HttpHeaders headers = tokenHeaderHelper.setupToken("INVALID_TOKEN");
+        ResponseEntity<?> response = trt.exchange("/products", HttpMethod.POST, new HttpEntity<>(request, headers), Void.class);
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenTokenIsExpired() {
+        token.setValidUntil(LocalDate.now().minusMonths(3));
+        tokenRepository.save(token);
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+    }
+
+    public ResponseEntity<?> setUpHeaderAndGetTheResponse() {
```

Comment:
create an overloaded version that takes the product id and make this one call that one with product.getId()